scripts/filter_expected_warnings.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+"""Script to filter out old warnings from doxygen."""
+
+import sys
+import re
+
+
+def compare_lists(list1, list2):
+    """Simple comparison of two lists."""
+    if len(list1) != len(list2):
+        return False
+    for (elt1, elt2) in zip(list1, list2):
+        if elt1 != elt2:
+            return False
+    return True
+
+
+class DoxygenWarning(object):
+    """Doxygen warning class."""
+
+    def __init__(self, firstline, filename, warning):
+        self.firstline = firstline
+        self.filename = filename
+        self.warning = warning
+        self.otherlines = []
+
+    def approx_equals(self, other):
+        """Return true if warnings have same filename and warning message."""
+        if self.filename != other.filename:
+            return False
+        if self.warning != other.warning:
+            return False
+        return compare_lists(self.otherlines, other.otherlines)
+
+    def is_unexpected(self, expected_warnings):
+        """Return true unless this is in the list of expected warnings."""
+        for expected in expected_warnings:
```

Comment:
Minor point that this leads to a quadratic time algorithm. Fortunately we don't have millions of warnings so this doesn't really matter.

Reply:
The script runs in <100ms on my laptop.